presto-snowflake/src/main/java/io/prestosql/plugin/snowflake/SnowflakeConfig.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.snowflake;
+
+import io.airlift.configuration.Config;
+
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+public class SnowflakeConfig
+{
+    private int fetchSize = 1000;
+
+    @NotNull
+    @Min(1)
+    public int getFetchSize()
+    {
+        return fetchSize;
+    }
+
+    @Config("snowflake.fetch-size")
```

Comment:
While it is tempting to have tuning knobs, we should add them only if needed.
Having a knob doesn't exempt us from picking a good default, and a good default is often all is needed.

Reply:
@findepi Agreed. I can remove it as I'm fairly agnostic on that parameter, but I added this following a suggestion from @Praveen2112.

Reply:
@Praveen2112 did you have any particular reason to add this for Snowflake?

Reply:
@findepi Agreed. Thought it might be better if we could have a way to configure it out

Reply:
@pgagnon But does snowflake JDBC connector uses the fetch size value ? 
https://community.snowflake.com/s/question/0D50Z00007gjDFVSA2/do-snowflake-jdbc-driver-support-statementsetfetchsize-i-am-getting-out-of-memory-error-on-selecting-from-table-having-large-number-of-rows-through-jdbc

I skimmed through their JDBC code and it looks like they are not using it ? (I might have missed its usage)

Reply:
Better -- because?



If this is needed today, for a use-case at hand that we can identify (name), **but** we cannot support automatically, then yes, we should have a knob. But we should not add a knob just "because it gives flexibility".

A knob just moves the responsibility for smart decisions onto the users / operators.
And, limits our possibility to do smart decisions ourselves in the future, when we learn how to make such decisions.